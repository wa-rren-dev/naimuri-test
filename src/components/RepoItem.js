import { Link } from "react-router-dom";
import { DateTime } from "luxon";
import s from "./RepoItem.module.scss";

export function RepoItem({ repo }) {
  const { name, description } = repo;
  return (
    <li className={s.RepoItem} aria-label={name}>
      <div>
        <h2>
          <Link to={name}>{name}</Link>
        </h2>
        {description && (
          <p className={s.RepoItem__description}>{description}</p>
        )}
        <dl className={s.RepoItem__meta}>
          <div className={s.RepoItem__metaItem}>
            <dt>
              <span
                aria-hidden="true"
                className={s.RepoItem__metaIcon}
                title="Stars"
              >
                ‚≠ê
              </span>
              Ô∏è <span className={s.RepoItem__metaLabel}>Stars</span>
            </dt>
            <dd>{repo.stargazers_count}</dd>
          </div>
          <div className={s.RepoItem__metaItem}>
            <dt>
              <span
                aria-hidden="true"
                className={s.RepoItem__metaIcon}
                title="Forks"
              >
                üç¥
              </span>{" "}
              <span className={s.RepoItem__metaLabel}>Forks</span>
            </dt>
            <dd>{repo.forks_count}</dd>
          </div>
          <div className={s.RepoItem__metaItem}>
            <dt>
              <span
                aria-hidden="true"
                className={s.RepoItem__metaIcon}
                title="Link to repository"
              >
                üîó
              </span>{" "}
              <span className={s.RepoItem__metaLabel}>Link</span>
            </dt>
            <dd>
              <a
                href={repo.html_url}
                aria-label={`View ${name} repo on Github`}
              >
                View on Github
              </a>
            </dd>
          </div>
          <div className={s.RepoItem__metaItem}>
            <dt>
              <span
                aria-hidden="true"
                className={s.RepoItem__metaIcon}
                title="Open issues"
              >
                üßê
              </span>{" "}
              <span className={s.RepoItem__metaLabel}>Open Issues</span>
            </dt>
            <dd>{repo.open_issues_count}</dd>
          </div>
          <div className={s.RepoItem__metaItem}>
            <dt>
              <span
                aria-hidden="true"
                className={s.RepoItem__metaIcon}
                title="Repo created"
              >
                üê£
              </span>{" "}
              <span className={s.RepoItem__metaLabel}>Created</span>
            </dt>
            <dd>{DateTime.fromISO(repo.created_at).toRelative()}</dd>
          </div>
        </dl>
      </div>
    </li>
  );
}
