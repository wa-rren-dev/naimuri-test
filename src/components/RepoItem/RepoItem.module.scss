@use "../App/App.scss" as App;

.RepoItem {
  background: white;
  border-top: 2px solid #ccc;
  padding-top: 2rem;
  margin-top: 2rem;

  &:first-of-type {
    margin-bottom: 0;
  }

  &__description {
    margin-bottom: 1rem;

    &:before {
      content: '\00201C';
    }

    &:after {
      content: '\00201D';
    }
  }

  &__meta {
    display: flex;
    align-items: center;
    justify-content: flex-start;
  }

  &__metaItem {
    display: flex;
    align-items: baseline;
    background: #efffff;
    border: 1px solid rebeccapurple;
    padding: 0.2rem 0.5rem;
    border-radius: 3px;
    margin-right: 1rem;

    &:last-of-type {
      margin-right: 0;
    }
  }

  &__metaIcon {
    margin-right: 0.5rem;
    transform: scale(2.25);
    display: inline-block;
    text-shadow: 2px 2px 0 rgba(0, 0, 0, 0.1);
  }

  &__metaLabel {
    @extend .visually-hidden;
  }
}
