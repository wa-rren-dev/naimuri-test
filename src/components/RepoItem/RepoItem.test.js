import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { RepoItem } from "./RepoItem";
import { MemoryRouter } from "react-router";

describe("RepoItem", () => {
  it("should render a link to the repo", async () => {
    const repo = {
      name: "test-repo",
      description: "A test repo",
      stargazers_count: 0,
      forks_count: 0,
      html_url: "http://example.com",
    };
    render(
      <MemoryRouter>
        <RepoItem repo={repo} />
      </MemoryRouter>
    );
    const link = screen.getByRole("link", {
      name: /View readme for test-repo/i,
    });
    expect(link).toBeInTheDocument();
  });
  it("should match the snapshot", () => {
    const repo = {
      name: "test-repo",
      description: "A test repo",
      stargazers_count: 0,
      forks_count: 0,
      html_url: "http://example.com",
    };
    const { asFragment } = render(
      <MemoryRouter>
        <RepoItem repo={repo} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
