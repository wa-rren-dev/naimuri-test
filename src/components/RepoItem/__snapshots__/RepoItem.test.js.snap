// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoItem should match the snapshot 1`] = `
<DocumentFragment>
  <li
    aria-label="test-repo"
    class="RepoItem"
  >
    <div>
      <h2>
        <a
          aria-label="View readme for test-repo"
          href="/test-repo"
        >
          test-repo
        </a>
      </h2>
      <p
        class="RepoItem__description"
      >
        A test repo
      </p>
      <dl
        class="RepoItem__meta"
      >
        <div
          class="RepoItem__metaItem"
        >
          <dt>
            <span
              aria-hidden="true"
              class="RepoItem__metaIcon"
              title="Stars"
            >
              ‚≠ê
            </span>
            Ô∏è 
            <span
              class="RepoItem__metaLabel"
            >
              Stars
            </span>
          </dt>
          <dd>
            0
          </dd>
        </div>
        <div
          class="RepoItem__metaItem"
        >
          <dt>
            <span
              aria-hidden="true"
              class="RepoItem__metaIcon"
              title="Forks"
            >
              üç¥
            </span>
             
            <span
              class="RepoItem__metaLabel"
            >
              Forks
            </span>
          </dt>
          <dd>
            0
          </dd>
        </div>
        <div
          class="RepoItem__metaItem"
        >
          <dt>
            <span
              aria-hidden="true"
              class="RepoItem__metaIcon"
              title="Link to repository"
            >
              üîó
            </span>
             
            <span
              class="RepoItem__metaLabel"
            >
              Link
            </span>
          </dt>
          <dd>
            <a
              aria-label="View test-repo repo on Github"
              href="http://example.com"
            >
              View on Github
            </a>
          </dd>
        </div>
        <div
          class="RepoItem__metaItem"
        >
          <dt>
            <span
              aria-hidden="true"
              class="RepoItem__metaIcon"
              title="Open issues"
            >
              üßê
            </span>
             
            <span
              class="RepoItem__metaLabel"
            >
              Open Issues
            </span>
          </dt>
          <dd />
        </div>
        <div
          class="RepoItem__metaItem"
        >
          <dt>
            <span
              aria-hidden="true"
              class="RepoItem__metaIcon"
              title="Repo created"
            >
              üê£
            </span>
             
            <span
              class="RepoItem__metaLabel"
            >
              Created
            </span>
          </dt>
          <dd />
        </div>
      </dl>
    </div>
  </li>
</DocumentFragment>
`;
